staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -558,7 +558,7 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		// the user matches the version of latest storage object.
 		resourceVersion, err := e.Storage.Versioner().ObjectResourceVersion(obj)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, kubeerr.NewConflict(qualifiedResource, name, fmt.Errorf("Failed to get object resource version %v", err))
```

Comment:
I don't think this is a conflict, it seems like a bad request

Reply:
I don't really understand what error this is handling here. `Failed to get object resource version %v` really sounds like a 500 (server screwed up). The scenario I described in the issue this PR is claiming to fix however is definitely a 409, neither a 500 nor a 400: https://github.com/kubernetes/kubernetes/issues/68202

Reply:
this line is returning an error related to failure to parse the resourceVersion on the incoming object from the user. if it cannot be parsed, that means the user submitted an object with a resourceVersion in a format unknown to the server, which is a bad request, not a conflict.

Reply:
Ah, that makes sense to me. I agree. I didn't realise there was anything to parse about the resourceVersion, I thought it was just a number.

So to clarify, this is _not_ for the case where a _valid_ but _old_ resource version was passed (that should be a 409).

Reply:
> I didn't realise there was anything to parse about the resourceVersion, I thought it was just a number.

resourceVersion is defined as an opaque string. It happens to be a number today.

> So to clarify, this is not for the case where a valid but old resource version was passed (that should be a 409).

That's right. This line is erroring if the format of the resourceVersion was incorrect.

Reply:
> I don't think this is a conflict, it seems like a bad request or unprocessable entity

Done, now returns bad request.